libkbfs/md_util.go

```diff
@@ -399,88 +399,57 @@ func getFileBlockForMD(ctx context.Context, bcache BlockCache, bops BlockOps,
 
 func reembedBlockChanges(ctx context.Context, codec kbfscodec.Codec,
 	bcache BlockCache, bops BlockOps, tlfID tlf.ID, pmd *PrivateMetadata,
-	rmdWithKeys KeyMetadata) error {
+	rmdWithKeys KeyMetadata, log logger.Logger) error {
 	info := pmd.Changes.Info
 	if info.BlockPointer == zeroPtr {
 		return nil
 	}
 
-	// Fetch the top-level block.
-	fblock, err := getFileBlockForMD(
-		ctx, bcache, bops, info.BlockPointer, tlfID, rmdWithKeys)
-	if err != nil {
-		return err
-	}
-
-	buf := fblock.Contents
-	if fblock.IsInd {
-		numFetchers := len(fblock.IPtrs)
-		if numFetchers > maxParallelBlockGets {
-			numFetchers = maxParallelBlockGets
-		}
-
-		type iptrAndBlock struct {
-			ptr   IndirectFilePtr
-			block *FileBlock
-		}
-
-		// Fetch all the child blocks in parallel.
-		iptrsToFetch := make(chan IndirectFilePtr, len(fblock.IPtrs))
-		indirectBlocks := make(chan iptrAndBlock, len(fblock.IPtrs))
-		eg, groupCtx := errgroup.WithContext(ctx)
-		fetchFn := func() error {
-			for iptr := range iptrsToFetch {
-				select {
-				case <-groupCtx.Done():
-					return groupCtx.Err()
-				default:
-				}
-
-				fblock, err := getFileBlockForMD(groupCtx, bcache, bops,
-					iptr.BlockPointer, tlfID, rmdWithKeys)
-				if err != nil {
-					return err
-				}
-
-				indirectBlocks <- iptrAndBlock{iptr, fblock}
-			}
-			return nil
-		}
-		for i := 0; i < numFetchers; i++ {
-			eg.Go(fetchFn)
-		}
-		for _, iptr := range fblock.IPtrs {
-			iptrsToFetch <- iptr
-		}
-		close(iptrsToFetch)
-		if err := eg.Wait(); err != nil {
-			return err
+	// Treat the unembedded block change like a file so we can reuse
+	// the file reading code.
+	file := path{FolderBranch{tlfID, MasterBranch},
+		[]pathNode{{info.BlockPointer, ""}}}
```

Comment:
maybe worth putting in "\<MD rev xx\>" for the path node name, in case it's logged somewhere?